tests/unittests/graphTest.cpp

```diff
@@ -460,3 +460,97 @@ TEST(Graph, disableUnrollingGroupConv) {
   EXPECT_GT(numberOfNodesOpenCL, numberOfNodesInterpreter);
 #endif // GLOW_WITH_OPENCL
 }
+
+/// Check that save nodes are properly scheduled.
+/// That is, they happen after the last use of the related variable.
+/// In that test, the order of the creation of the nodes give a valid schedule.
+TEST(Graph, schedulingOfSavesOrderProvided) {
+  ExecutionEngine EE;
+
+  Tensor inputs(ElemKind::FloatTy, {1, 32, 32, 3});
+
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("interpret");
+  auto *A = mod.createVariable(ElemKind::FloatTy, {3, 32}, "A",
+                               VisibilityKind::Public,
+                               Variable::TrainKind::Xavier, 1.0);
+  auto *zero = mod.createVariable(A->getType(), "zero", VisibilityKind::Public,
+                                  Variable::TrainKind::Broadcast, 0.0);
+
+  auto *B = mod.createVariable(A->getType(), "B", VisibilityKind::Public,
+                               Variable::TrainKind::Xavier, 1.0);
+
+  auto *addAB = F->createAdd("addAB", A, B);
+
+  auto *saveNode = F->createSave("ret", addAB);
+  F->createSave("resetA", zero, A);
+
+  // Copy the value of A.
+  Tensor AOrig = A->getPayload().clone();
+
+  EE.compile(CompilationMode::Infer, F);
+  EE.run({}, {});
+  auto *ret = saveNode->getVariable();
+  auto handleAOrig = AOrig.getHandle<>();
+  auto handleA = A->getPayload().getHandle<>();
+  auto handleB = B->getPayload().getHandle<>();
+  auto handleRet = ret->getPayload().getHandle<>();
+  bool allEqual = true;
+  for (unsigned row = 0; row != 3; ++row) {
+    for (unsigned column = 0; column != 32; ++column) {
+      EXPECT_EQ(handleA.at({row, column}), 0.0);
+      allEqual &= handleAOrig.at({row, column}) + handleB.at({row, column}) ==
+                  handleRet.at({row, column});
+    }
+  }
+  EXPECT_TRUE(allEqual);
+}
+
+/// Same as the previous test, except the order in which the nodes
```

Comment:
nit: maybe say `Same as schedulingOfSavesOrderProvided` in case ordering of tests changes.